### ISSUE TITLE
unnecessary else after return/raise [considered to disable pylint rule]

### ISSUE THREAD

**Issue by aleksandr-kotlyar** [yashaka/selene#271]
### Todo
Decide should we use `"else"` after `"return"` and `"else"` after `"raise"` in selene or make a rule to not allow this.
Example in method:
https://github.com/yashaka/selene/blob/b24e85e0d085ae8675f512cc5654b368ac8edec7/selene/core/entity.py#L138-L153
It seems that our eyes should end reading the function on "return" and we should ignore all after. But when we see "else" after "return" we stop and start thinking: "oh, there is 'else', what's here?" 🙂 And loosing context. 

### Suggested change

Before
```
if cached.matching(element_is_present):
    return TimeoutException(
        error.msg
        + f'\nActual webelement: {query.outer_html(element)}')
else:
    return error
```
After
```
if cached.matching(element_is_present):
    return TimeoutException(
        error.msg
        + f'\nActual webelement: {query.outer_html(element)}')
return error
```
I believe that it will be easer to read without "else".
I suggest we should spread this changes to the whole project, add convention rule to Code conventions and lint it's violations automatically.
The final argument to accept the rule: there are only 5 violations in the code for now - good chance to fix them all together.

**Comment by yashaka** [comment]
the structure of this styl

```
if cached.matching(element_is_present):
    return TimeoutException(
        error.msg
        + f'\nActual webelement: {query.outer_html(element)}')
else:
    return error
```

is more "aligned", this gives some kind of simplicity
we read the code and come to the place where there is a branch in logic
in order to deal with this code - it's important to see this branching at once
and such style show this in a more straightforward way


When comparing to this...

```
if cached.matching(element_is_present):
    return TimeoutException(
        error.msg
        + f'\nActual webelement: {query.outer_html(element)}')
return error
```

Here the "branching" is hidden. 
In fact, the original argument against the first version: "oh, there is 'else', what's here?" - in fact shows an advantage of the first version in context of "highlighting the branching aspect of the logic". This is actually very good that you got this question and will think about it:)


Such "explicit branching style" of the first version comes from Functional Style of programming where it would a common way of solving things. Yet in functional style the return will be before `if`, not after. 

But in our code we have two returns - and this may confuse a bit... someone... 

In some cases, when we do a validation of function parameters it's pretty convenient to return immediately at the beginning, or throw exception

In other cases to much of returns spreaded all over the code inside fn, may make things more complicated.

it depends...

we can refactor the original code also in this way:

```
return error if not cached.matching(element_is_present)
	else TimeoutException(
        error.msg
        + f'\nActual webelement: {query.outer_html(element)}')
```

And for me it seems to better than both previous ones... yet emphesizing the branching aspect

But should we put the suggested rule? not sure...
we can try to refactor all cush 5 cases to the something like the latter, add a rule, and then see will we met some cases where if else style of branching would be better...

**Comment by yashaka** [comment]
Hm...
But here...

![image](https://user-images.githubusercontent.com/1961118/104122407-842c1200-534d-11eb-9dc9-ffc8ae5f363e.png)

the `something if predicate else something_else` style will not work... 
and yet I believe that it's important to emphesize the branching in such context.
Yes, while reading such code we becomve slower... We have to think on why this code is inside else, and check the main if clause, and understand it – but, in fact, this is the goal – to slow down, understand why this branching exist, and only after that proceed any further refactoring, etc.

Let's close this issue, not adding this linter rule, and refactor only those code snippets where we can switch from `if ... return ... else return ...` to `return ... if ... else ...`

**Comment by aleksandr-kotlyar** [comment]
Ok, i will make new PR with this rule disabled in `.pylintrc` config as agreed.
